test/OpenTelemetry.Tests/Metrics/AggregatorTestsBase.cs

```diff
@@ -200,11 +202,11 @@ public void MultiThreadedHistogramUpdateAndSnapShotTest()
         };
 
         var numberOfThreads = 2;
-        var snapshotThread = new Thread(HistogramSnapshotThread);
+        var snapshotThread = new Thread(HistogramSnapshotThread!);
```

Comment:
This is strange to me that we would need to declare that a method is not-null.

The error message:
> CS8622: Nullability of reference types in type of parameter 'obj' of 'void AggregatorTestsBase.HistogramSnapshotThread(object obj)' doesn't match the target delegate 'ParameterizedThreadStart' (possibly because of nullability attributes).

Looking up ParametirizedThreadStart:
https://learn.microsoft.com/en-us/dotnet/api/system.threading.parameterizedthreadstart?view=net-8.0
> public delegate void ParameterizedThreadStart(object? obj);

and comparing that to the HistogramSnapshotThread method:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/cc0b9e9d0f874d4cbafd801d4af343f02e4ffbfe/test/OpenTelemetry.Tests/Metrics/AggregatorTestsBase.cs#L469

@ysolomchenko, I think instead we should change the parameter in the `HistogramSnapshotThread` to be `object?`. Can you please make the change to this and the `HistogramUpdateThread` method as well?

Reply:
fixed in e2f959e4dc14ba4a52cc1ff991d620085aa62ccc